### PR TITLE
Add CSpell checker to CI and fix typos

### PR DESCRIPTION
**Pull request by jochem-brouwer**
This PR adds `cspell` to the CI. To use it locally, from the repository root, use `npm run cspell`. This will spell-check all *.ts files in the packages. It will also check the markdown, but it will not check the markdowns in the docs.

The output of cspell will give context and - if possible - also give word suggestions how to fix it.

If a word is clearly correct but cspell flags it, add it to the words list in `config/cspell-md.json` (for markdown) or `config/cspell-ts.json`. There are two jsons, one for scripts and one for the md files.

If a word is intentionally incorrect, but it should be ignored, add ` // cspell:disable-line` to a script or `<!--- cspell:disable-line --->` to a markdown file.

The CI will trigger the `npm run cspell` job, if it finds any mistakes, the CI will throw.

- [x] Add and fix markdown
- [x] Add to CI
- [x] Test CI fails if cspell throws

** To review this **:

Please check the `config/cspell-md.json` and the `config/cspell-ts.json` for any words which I have marked as correct, but they are still wrong. Some words might need context on why these are added, to check: remove the file from the json, save the file, and then run `npm run cspell:ts` for scripts or `npm run cspell:md` for markdown to see the context in which this word is found.

### CODE REVIEW

**Review by holgerd77 — CHANGES_REQUESTED** [review]
Really important: please do not merge this yet respectively do not run the execution part (the doc changes), otherwise integrating the release PR #3527 changes into `master` will be a total merge hell!!

(I mentioned this kind of problem set a couple of times in the chat, actually I am getting a bit tired to always need to play the last guard resort to look for these kind of things 😳)

Apart from this base complaint this is really a great PR though! 😄

**Comment by jochem-brouwer** [comment]
I think I got a little bit too enthusiastic, because I was aware of this (this is also the reason why I have split up the md/ts files and have not committed any typedoc changes). Have marked PR as draft so it also cannot be accidentally merged.

**Comment by holgerd77** [comment]
Haha, then maybe just mention it and mark as "blocked" to reduce my 😱 moments! 😂

**Comment by jochem-brouwer** [comment]
Once we merge this, I suggest the spell check job is added to the required CI jobs.

File: config/README.md
Comment (gabrocheleau) [inline config/README.md:26]: This is wrong

File: packages/block/src/from-beacon-payload.ts
Comment (gabrocheleau) [inline packages/block/src/from-beacon-payload.ts:1]: Why is this necessary?

File: packages/block/test/from-rpc.spec.ts
Comment (gabrocheleau) [inline packages/block/test/from-rpc.spec.ts:1]: Not a big fan of having all these new `cspell:ignore` lines at the beginning of files just for spell checking. Can't we add all of these terms to the config file?

File: packages/client/CHANGELOG.md
Comment (gabrocheleau) [inline packages/client/CHANGELOG.md:173]:
```suggestion
- Fixes regarding beacon sync, vmexecution and further log improvements, PR [#3094](https://github.com/ethereumjs/ethereumjs-monorepo/pull/3094)
```

File: packages/block/src/from-beacon-payload.ts
Comment (jochem-brouwer) [reply packages/block/src/from-beacon-payload.ts:1]: It is a variable name used in the file

File: packages/trie/scripts/view.ts
Comment (gabrocheleau) [inline packages/trie/scripts/view.ts:26]: Is this update intentional? Not sure what the resulting type is, but seems it could be different

File: config/README.md
Comment (jochem-brouwer) [reply config/README.md:26]: Oof, this must be caused by a find-and-replace, good find

File: packages/trie/scripts/view.ts
Comment (gabrocheleau) [reply packages/trie/scripts/view.ts:26]: Also not sure if the formatting changes later in the file are intentional

Comment (jochem-brouwer) [reply packages/trie/scripts/view.ts:26]: Not sure, I do not recall changing this. Cspell should not change anything, so this could be caused by the linter (for some reason?)

File: packages/util/CHANGELOG.md
Comment (gabrocheleau) [inline packages/util/CHANGELOG.md:80]:
```suggestion
- Performance: New BigInt constants (`BIGINT_0`, `BIGINT_32`, `BIGINT_2EXP96`,...) in the `bytes` module for re-usage along performance optimizations, PR [#3050](https://github.com/ethereumjs/ethereumjs-monorepo/pull/3050)
```

**Comment by holgerd77** [comment]
I didn't know we had THAT many spelling mistakes in our docs 🤯 (so many POAP opportunities lost 😂 ).

File: config/README.md
Comment (jochem-brouwer) [reply config/README.md]: Have reverted back to correct URL!

File: packages/trie/scripts/view.ts
Comment (jochem-brouwer) [reply packages/trie/scripts/view.ts:26]: Have reverted

File: packages/block/test/from-rpc.spec.ts
Comment (scorbajio) [reply packages/block/test/from-rpc.spec.ts:1]: Renaming the variables (wtxns -> withTxns) may be the best solution if the variable names are not very intuitive. Otherwise, it would make sense to ignore by file or line, but then we'd have to keep aware of what unique variable names are coming up in frequency of usage, so that we could eventually switch over to adding them to the config file instead of ignoring them on a per line or per file basis.

**Review by scorbajio — COMMENTED** [review]
Really useful addition to the CI 🙂

**Comment by jochem-brouwer** [comment]
@scorbajio See #3599, I did those renames there :) I think I will actually merge that one into this PR before moving it into master (after I've finished the CI conflicts)

**Comment by jochem-brouwer** [comment]
Hi all, I have addressed concerns in #3599 and just squash-merged it in this commit: https://github.com/ethereumjs/ethereumjs-monorepo/pull/3590/commits/5167780d77c9f95cf2679937459bbb65d38fbf63

I think this will resolve most/all concerns/issues?

File: packages/block/src/from-beacon-payload.ts
Comment (jochem-brouwer) [reply packages/block/src/from-beacon-payload.ts]: This is now removed!

File: packages/block/test/from-rpc.spec.ts
Comment (jochem-brouwer) [reply packages/block/test/from-rpc.spec.ts]: Have addressed -> https://github.com/ethereumjs/ethereumjs-monorepo/pull/3590#issuecomment-2294846162

**Review by gabrocheleau — APPROVED** [review]

**Comment by jochem-brouwer** [comment]
Thanks for all the reviews :+1: :smile:

**Comment by holgerd77** [comment]
Looking at this once again, and this is really monumental! 🤯 

One of the big structural updates for the whole monorepo! 💯 🎉

**Pull request merged**
